frontend-react/src/hooks/UseLookupTable.ts

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+import {
+    lookupTableApi,
+    LookupTable,
+    ValueSet,
+    LookupTables,
+} from "../network/api/LookupTableApi";
+
+export const generateUseLookupTable =
+    <T extends unknown>(tableName: LookupTables) =>
+    () => {
+        return useLookupTable<T>(tableName);
+    };
+
+export async function GetLatestVersion(
+    tableName: LookupTables
+): Promise<number> {
+    let response;
+    try {
+        response = await axios(lookupTableApi.getTableList()).then(
+            (response) => response.data
+        );
+
+        let filteredBody: LookupTable[] = response.filter(
+            (tv: LookupTable) => tv.tableName === tableName && tv.isActive
+        );
+        if (filteredBody.length === 0) {
```

Comment:
I'm a little confused about the purpose of this piece of logic. It seems like in the end we have a list of versions where tableNames match, regardless of whether they are active, with no indication of whether active or inactive versions are being used. Can we just skip the `isActive` check?

Reply:
@doug-s-nava I'd like to (and almost did yesterday), but we're trying to address a little fuzziness in regards to how the API handles active/inactive state. We were told that when creating new versions of a lookup table (via updating the table) that the new version would be by default _inactive_, so that OnO could control exactly when the table version goes "live" (via a separate CLI+API call). But the way it seems to play out in reality is: when you create a new version, that new version is automatically activated. Perhaps someone decided along the way that that extra call wasn't needed after all etc? Unknown.

Regardless, the ability to inactivate tables exists (though it wasn't accounted for in the designs) and is probably only used in emergency rollback scenarios. So what we're trying to do here is prioritize grabbing the latest _active_ version, and in the (hopefully) rare case that there _are_ no active versions- at least display the latest _inactive_ version and visually indicate that on the page somehow (TBD)

Now that I've written all that, however, I'm wondering if it's actually better (in light of all this ambiguity) to really just restrict everything to **active only** for the time being. Why don't we add this to our To do list for the details page work (#5442) in the interests of moving this along...

Reply:
I am fine with moving this to a follow up ticket! can you add a todo in the code, or update the ticket so we don't forget to revisit?

this is definitely a little more complicated than I appreciated. It might be that there is a slightly more elegant way to do what you're proposing. Or maybe there isn't!